ecdsa.go

```diff
@@ -162,48 +162,35 @@ func unmarshalEcParams(b []byte) (elliptic.Curve, error) {
 	return nil, ErrUnsupportedEllipticCurve
 }
 
-func unmarshalEcPoint(b []byte, c elliptic.Curve) (x *big.Int, y *big.Int, err error) {
-	// Decoding an octet string in isolation seems to be too hard
-	// with encoding.asn1, so we do it manually. Look away now.
-	if b[0] != 4 {
+func unmarshalEcPoint(b []byte, c elliptic.Curve) (*big.Int, *big.Int, error) {
+	var pointBytes []byte
+	extra, err := asn1.Unmarshal(b, &pointBytes)
+	if err != nil {
 		return nil, nil, ErrMalformedDER
```

Comment:
Given we now have an error from asn1 Unmarshal we should return that.

Reply:
In line with our conversations yesterday, there are no exported errors any more. This particular bit of code now returns the underlying error (with decoration).